grammar.ebnf

```diff
@@ -5,16 +5,23 @@ entry                ::= comment
                       ;
 
 comment              ::= '#' .*;
-section              ::= '[[' keyword ']]';
+section              ::= '[[' __ keyword __ ']]';
+
 __                   ::= [ \t]*;
 NL                   ::= [\r\n]+;
+identifier-head      ::= [a-zA-Z_.?-];
+identifier-tail      ::= [a-zA-Z0-9_.?-];
+keyword-head         ::= [^=|#/{}\[\]()0-9];
+keyword-tail         ::= [^=|#/{}\[\]() ];
+keyword-last         ::= [^=|#/{}\[\]()];
 
-identifier           ::= [a-zA-Z_.?-] ([a-zA-Z0-9_.?-])*;
+identifier           ::= identifier-head (identifier-tail)*;
 variable             ::= '$' identifier;
-keyword              ::= [^=|#{}\[\]()]+;
+keyword              ::= keyword-head (keyword-tail* keyword-last)?
```

Comment:
I'm concerned that the following will look cryptic:

    keyword ::= [^=|#/{}\[\]()0-9] ([^=|#/{}\[\]() ]* [^=|#/{}\[\]()])?;

Then again, the grammar is intended at parsers not humans, so maybe that's okay. I would leave it in the clearer form at least for now since we don't use this grammar yet to generate parsers which in turn would allow us to fix bugs in the grammar itself.  Thus readable-by-human is still pretty important to me.